src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -683,6 +684,16 @@ protected virtual bool TryLoadFromCache(JsonOperationContext context)
                                 }
 
                                 response = await preferredTask.ConfigureAwait(false);
+                                if (sessionInfo?.GetLastClusterTransactionFunc?.Invoke() != null)
```

Comment:
This is a bad idea. You are relying on mutable global state that doesn't belong to you.
You might be reading some _other_ command state, for example.